File: Cpp/883_Projection_Area_of_3D_Shapes.cpp
class Solution {
public:
    int projectionArea(vector<vector<int>>& grid) {
        int result = 0;
        int n = grid.size();
        for(int i=0; i<n; ++i){
            int xz = 0, yz = 0;
            for(int j=0; j<n; ++j){
                result += grid[i][j] > 0 ? 1 : 0;
                xz = max(xz, grid[i][j]);
                yz = max(yz, grid[j][i]);
            }
            result += xz + yz;
        }
        
        return result;
    }
};

File: Kotlin/883_Projection_Area_of_3D_Shapes.kt
class Solution {
    fun projectionArea(grid: Array<IntArray>): Int {
        var result = 0
        var n = grid.size
        
        for(i in 0 until n){
            var xz = 0
            var yz = 0
            for(j in 0 until n){
                result += if(grid[i][j] > 0) 1 else 0
                xz = maxOf(xz, grid[i][j])
                yz = maxOf(yz, grid[j][i])
            }
            result += xz + yz
        }
        
        return result
    }
}